utils/das_data_sorter.py

```diff
@@ -136,6 +137,53 @@ def gps_data(self, data:pd.Series) -> pd.Series:
         
         return new_gps_data
     
+    def smooths_data(self, data:pd.Series, technique:str, n:int=3) -> pd.Series:
+        if n < 3 or n > len(data):
+            raise ValueError("Number of smoothing points must be at least 3 and less than the length of the data set to perform smoothing.")
+        smooth_data_array = [] 
+
+        if technique == "mean" and n % 2 != 0: # Mean Smoothing for odd number N
```

Comment:
Will definitely separate mean_smoothing and median_smoothing! Forgot, we mentioned that last time too. 

Although, I think an odd-numbered N pathway and an even-numbered N pathway should remain separate. It's because the even-numbered pathway has an extra step where the averaged data points is 'centered'. In the video, for a 3-point mean smoothing, the averaged data point was already centered with a corresponding time. Whereas for a 4-point mean smoothing, the average data point was in-between numbers, so an extra step of 'centering' was done.

I actually had the same idea about putting an if statement to check for an even number N, but it wouldn't work out (unless you think of something, please let me know). 

Basically, if I stored the averaged data points into `smooth_data_array`, then I can take the values from the same array to do centering and then update the values in the same array (without overwriting something needed). Sounds like it works. 

However, the problem is that 'centering' reduces the number of data points, so you won't have the same number of data points. This means that when centering is finished, you will have leftover data points that are not part of the final converted data. This leftover data points could be a random spike or part of the original data. Because of this issue and the extra steps, I was thinking of separating them. 

If there's time tomorrow, I can demonstrate with a diagram during/after the meeting, if unclear.